near-sdk-core/src/code_generator/item_impl_info.rs

```diff
@@ -657,8 +657,36 @@ mod tests {
         let expected = quote!(
                 #[cfg(not(target_arch = "wasm32"))]
                 pub fn method(&self, k: String,) -> near_sdk::PendingContractTx {
-                    let args = near_sdk::serde_json::json!({ "k" : k });
-                    near_sdk::PendingContractTx::new(& self . account_id, "method", args, true)
+                  let args = near_sdk::serde_json::json!({ "k": k })
+                  .to_string()
+                  .into_bytes();
+                  near_sdk::PendingContractTx::new_from_bytes(&self.account_id, "method", args, true)
+                }
+        );
+        assert_eq!(expected.to_string(), actual.to_string());
+    }
+
+    #[test]
+    fn marshall_borsh() {
```

Comment:
Can you also add a test for using `#result_serializer(borsh)`

Reply:
This isn't needed because the macros for the simulator transform methods to return a `PendingContractTx`.  Thus the result serialization happens when the contract gets run and then the `ExecutionResult` takes care of deserializing.